lib/java_buildpack/container/tomee/tomee_instance.rb

```diff
@@ -34,6 +44,21 @@ def tomcat_7_compatible
         @version < TOMEE_7
       end
 
+      private
+
+      def web_inf_lib
+        if ear?
+          # copy additional libraries to tomee lib folder
+          tomcat_lib
```

Comment:
Hi,

After a discussion with TomEE committers [1] it appeared that TomEE classloader has to be used as TomEE tries to instantiate the driver at a stage when there is no EAR/WAR classloaders.
There is a `Lazy` feature which instructs TomEE to delay the instantiation but this is not a solution when the application uses JPA.

What do you think about the following solution:
1. Use case - application wants to provide its own driver
- Application puts the driver under `EAR/drivers` folder
- Buildpack check that there is a driver in the application's binaries and does not download the driver jar file [2]
2. Use case - application does not package a driver with its binaries
- Buildpack check that there isn't a driver in the application's binaries and downloads the driver jar file [3]
- The driver is added to the additional libraries [4]
- Additional libraries are linked to the `EAR/drivers` folder

For both use cases the `common.loader` (see `TomEE/conf/catalina.properties`) has to be extended with the `EAR/drivers/*.jar`
`common.loader="${catalina.base}/lib","${catalina.base}/lib/*.jar","${catalina.home}/lib","${catalina.home}/lib/*.jar",${EAR.location}/drivers/*.jar`  

Regards,
Violeta

[1] http://tomee-openejb.979440.n4.nabble.com/Driver-location-in-an-EAR-file-td4681675.html
[2] https://github.com/cloudfoundry-community/tomee-buildpack/blob/master/lib/java_buildpack/framework/maria_db_jdbc.rb#L46
[3] https://github.com/cloudfoundry-community/tomee-buildpack/blob/master/lib/java_buildpack/framework/maria_db_jdbc.rb#L28
[4] https://github.com/cloudfoundry-community/tomee-buildpack/blob/master/lib/java_buildpack/framework/maria_db_jdbc.rb#L29